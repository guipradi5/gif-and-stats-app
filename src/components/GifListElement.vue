<template>
  <li @click="emit('select', element)">{{ element.content_description }}</li>
</template>

<script setup>
import { ref, defineProps, defineEmits } from "vue";

const props = defineProps(["element"])
const element = ref(props.element)

const emit = defineEmits(["select"])
</script>

<script>
export default {
  name: "GifListElement"
}
</script>

<style scoped lang="scss">
li {
  padding: 10px 20px;
  list-style: none;
  cursor: pointer;
}
</style>
