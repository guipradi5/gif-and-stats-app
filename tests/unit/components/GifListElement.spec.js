import { shallowMount } from "@vue/test-utils";
import GifListElement from "@/components/GifListElement.vue";

const defaultProps = {
  element: {content_description: "Desc"}
}
const component = (props) => {
  return shallowMount(GifListElement, {
    props: props || defaultProps
  })
}

describe("GifListElement.vue", () => {
  it('GifListElement should mount', () => {
    const wrapper = component()
    expect(wrapper.vm).toBeTruthy()
  })

  it('GifListElement should render', async () => {
    const wrapper = await component()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('GifListElement should emit event', async () => {
    const wrapper = await component()
    wrapper.vm.$emit('select', defaultProps.element)
    expect(wrapper.emitted().select).toBeTruthy()
    expect(wrapper.emitted().select[0]).toEqual([defaultProps.element])

  })
});
