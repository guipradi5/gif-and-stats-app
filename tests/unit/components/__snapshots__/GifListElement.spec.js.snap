// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifListElement.vue GifListElement should render 1`] = `<li>Desc</li>`;
